### PR TITLE
Added custom mamba op and fix the mamba cache issue

### PR DESCRIPTION
**Pull request by zzhang37**
# What does this PR do?

Added custom mamba pscan op and fixed the mamba cache issue

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by libinta — COMMENTED** [review]
@zzhang37 please update test to reflect the change

File: optimum/habana/transformers/models/mamba/modeling_mamba.py [hunk 1]
```diff
@@ -1,16 +1,76 @@
...
+from transformers.models.mamba.configuration_mamba import MambaConfig
+from transformers.activations import ACT2FN
 from transformers.utils import (
     ModelOutput,
     logging,
 )
 
+from pathlib import Path
+import os
+base_dir = "/workspace/custom_op_pscan_all"
```
Comment (ssarkar2) [inline optimum/habana/transformers/models/mamba/modeling_mamba.py:16]: @libinta what would be the best way to set this without hardcoding?
Atleast an env var?

File: optimum/habana/transformers/models/mamba/modeling_mamba.py [hunk 2]
```diff
@@ -94,3 +154,277 @@ def gaudi_MambaForCausalLM_prepare_inputs_for_generation(
...
     )
     return model_inputs
+
+class gaudi_MambaMixer(nn.Module):
+    """
+    Compute ∆, A, B, C, and D the state space parameters and compute the `contextualized_states`.
+    A, D are input independent (see Mamba paper [1] Section 3.5.2 "Interpretation of A" for why A isn't selective)
+    ∆, B, C are input-dependent (this is a key difference between Mamba and the linear time invariant S4,
+    and is why Mamba is called **selective** state spaces)
+    """
```
Comment (ssarkar2) [inline optimum/habana/transformers/models/mamba/modeling_mamba.py:164]: @zzhang37  can you plz add a comment in the code about the different between this and original impl?

**Comment by jiminha** [comment]
@zzhang37  Also, all synapse dependencies merged in to 1.19 release?

File: optimum/habana/transformers/models/mamba/modeling_mamba.py (hunk 1 above)
Comment (regisss) [reply optimum/habana/transformers/models/mamba/modeling_mamba.py:16]: Is this dir generated on the fly? Or is it supposed to be downloaded (e.g. as part of an example) ?

Comment (zzhang37) [reply optimum/habana/transformers/models/mamba/modeling_mamba.py:16]: We can change to
my_dir = os.path.realpath(__file__)
my_len = my_dir.rfind("/")
base_dir = my_dir[:my_len]

File: optimum/habana/transformers/models/mamba/modeling_mamba.py (hunk 2 above)
Comment (zzhang37) [reply optimum/habana/transformers/models/mamba/modeling_mamba.py:166]: Done

**Review by libinta — COMMENTED** [review]
can you update readme front page, text-gen to add cmd and test case?

**Comment by regisss** [comment]
@zzhang37 What's the difference between this PR and #1573 ?

**Comment by zzhang37** [comment]
@regisss This PR is no longer needed. Please remove it. Thanks. We only need PR #1573.

**Pull request closed without merging**